resources/views/smartxp/protopolis_screen.blade.php

```diff
@@ -0,0 +1,328 @@
+                                activityDiv.appendChild(locationDiv)
+                                activityDiv.innerHTML += '<br>'
+                                activityDiv.innerHTML += '<hr>'
+
+
+                              timetable.appendChild(activityDiv)
+                              count++
+                          }
+                      })
+                      if (count === 0) {
+                          timetable.innerHTML = '<div class="notice">No more lectures today!</div>'
+                      }
+                  } else {
+                      timetable.innerHTML = '<div class="notice">No lectures today!</div>'
+                  }
+                  setTimeout(updateTimetable, 60000);
+              })
+              .catch(err => {
+                  console.error(err)
+                  timetable.innerHTML = '<div class="notice">Lectures could not be found...</div>'
+                  setTimeout(updateTimetable, 5000);
+              })
+      }
+
+      updateTimetable();
+
+      function updateActivities() {
+          get('{{ route('api::events::upcoming', ['limit' => 5]) }}')
+              .then(data => {
+                  if (data.length > 0) {
+                      document.getElementById("activities").innerHTML = '';
+                      data.forEach((activity) => {
+                          let start = moment.unix(activity.start)
+                          let end = moment.unix(activity.end)
+                          let time
+                          if (start.format('DD-MM') === end.format('DD-MM')) {
+                              time = start.format("DD-MM, HH:mm") + ' - ' + end.format("HH:mm")
+                          } else {
+                              time = start.format("DD-MM, HH:mm") + ' - ' + end.format("DD-MM, HH:mm")
+                          }
+                          let newDiv=document.createElement("div")
+                          newDiv.className="activity bg-img protubecard flex-grow-1"+ (activity.current ? "current" : (activity.over ? "past" : ""))
+                          newDiv.style.padding='15px'
+                          if(activity.image) {
+                              newDiv.style.background = 'linear-gradient(rgba(255,255,255,0.7), rgba(255,255,255,0.7)), url('+activity.image+')'
+                              newDiv.style.backgroundSize = 'cover'
+                              newDiv.style.backgroundPosition = 'center center'
+                              newDiv.style.backgroundRepeat = 'no-repeat'
+                          }
+
+
+                          let titleDiv=document.createElement("div")
+                          titleDiv.className="overflow-hidden"
+
+                          let titleSpan=document.createElement("span")
+                          titleSpan.innerHTML=activity.title
+                          titleSpan.className="text-nowrap text-truncate w-100 me-5 font-weight-bold font-size-lg"
```

Comment:
I think a scrolling title would be preferable. See the ProTube screen for an example.

Reply:
Implemented :)